cdc/owner.go

```diff
@@ -508,6 +507,7 @@ func (o *Owner) loadChangeFeeds(ctx context.Context) error {
 			// don't need to return an error.
 			log.Warn("create changefeed failed, retry later",
 				zap.String("changefeed", changeFeedID), zap.Error(err))
+			newCf.Close()
```

Comment:
What if it fails early in `err2 := o.etcdClient.SaveChangeFeedInfo(ctx, cfInfo, changeFeedID)`? Does it leak changeFeed?